File: cpp/Map/MapScriptBasicMaterial.c
// Type: Map.MapScriptBasicMaterial
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Map/MapScriptBasicMaterial.cs
// Prior real C# source (older reference): Assets/Scripts/Map/MapScript/Materials/MapScriptBasicMaterial.cs
// --------------------------------

// Map.MapScriptBasicMaterial$$.ctor
// il2cpp: void Map_MapScriptBasicMaterial___ctor (Map_MapScriptBasicMaterial_o* __this, const MethodInfo* method);
// 0x40b4c30

void Map_MapScriptBasicMaterial___ctor(Map_MapScriptBasicMaterial_o *__this,MethodInfo *method)

{
  MethodInfo *method_00;
  
  if (g_data_057ac4ee == '\0') {
    il2cpp_runtime_helper_023445d0(&"Misc/None");
    g_data_057ac4ee = '\x01';
  }
  method_00 = "Misc/None";
  (__this->fields).Texture = (System_String_o *)"Misc/None";
  il2cpp_runtime_helper_022b4080(&(__this->fields).Texture);
  if (g_data_057a9c86 == '\0') {
    il2cpp_runtime_helper_023445d0(&TypeInfo_Vector2);
    g_data_057a9c86 = '\x01';
  }
  (__this->fields).Tiling.fields = *(UnityEngine_Vector2_Fields *)(*(long *)(TypeInfo_Vector2 + 0xb8) + 8);
  if (g_data_057a65d5 == '\0') {
    il2cpp_runtime_helper_023445d0(&TypeInfo_Vector3);
    g_data_057a65d5 = '\x01';
  }
  (__this->fields).Offset.fields = **(UnityEngine_Vector2_Fields **)(TypeInfo_Vector3 + 0xb8);
  Map_MapScriptBaseMaterial___ctor((Map_MapScriptBaseMaterial_o *)__this,method_00);
  return;
}

File: csharp_body_deref/Scripts/Map/MapScriptBasicMaterial.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Map.MapScriptBasicMaterial
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Map/MapScriptBasicMaterial.c
// Prior real C# source (older reference): Assets/Scripts/Map/MapScript/Materials/MapScriptBasicMaterial.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;
using UnityEngine;
using Utility;

namespace Map;

[Token(Token = "0x200027E")]
public class MapScriptBasicMaterial : MapScriptBaseMaterial
{
	[Token(Token = "0x4000D8D")]
	[FieldOffset(Offset = "0x20")]
	[Order(3)]
	public string Texture;

	[Token(Token = "0x4000D8E")]
	[FieldOffset(Offset = "0x28")]
	[Order(4)]
	public Vector2 Tiling;

	[Token(Token = "0x4000D8F")]
	[FieldOffset(Offset = "0x30")]
	[Order(5)]
	public Vector2 Offset;

	[Token(Token = "0x6000F50")]
	[Address(RVA = "0x3FB4C30", Offset = "0x3FB4C30", VA = "0x3FB4C30")]
	public MapScriptBasicMaterial()
	{
	}
}
